wai-aria/role/contextual-roles.html

```diff
@@ -0,0 +1,102 @@
+<!doctype html>
+<html>
+<head>
+  <title>Contextual Role Verification Tests</title>
+  <script src="/resources/testharness.js"></script>
+  <script src="/resources/testharnessreport.js"></script>
+  <script src="/resources/testdriver.js"></script>
+  <script src="/resources/testdriver-vendor.js"></script>
+  <script src="/resources/testdriver-actions.js"></script>
+  <script src="/wai-aria/scripts/aria-utils.js"></script>
+</head>
+<body>
+
+<p>Verifies Required Context Roles from <a href="https://www.w3.org/TR/wai-aria-1.2/#scope">5.2.7 Required Context Role</a></p>
+<p>Also verifies Required Owned Roles from <a href="https://www.w3.org/TR/wai-aria-1.2/#mustContain">5.2.6 Required Owned Role</a></p>
+
+<!-- Notes for "required context" testing:
+
+- See Computed Role for more details on role computation when an element lacks required context: 
+https://w3c.github.io/core-aam/#roleMappingComputedRole 
+
+- See ARIA 1.2 spec for full listing of "Required Context Role" for each element:
+https://www.w3.org/TR/wai-aria-1.2/
+
+Identified roles with "Required Context" excluding abstract roles (e.g., child -> parent):
+
+- caption -> {figure, grid, table, treegrid}
+- cell -> row
+- columnheader -> row
+- gridcell -> row
+- listitem -> {list, directory} *Note: directory role is deprecated as of ARIA 1.2
+- menuitem -> {group, menu, menubar}
+- menuitemcheckbox -> {group, menu, menubar}
+- menuitemradio -> {group, menu, menubar}
+- option -> {group, listbox}
+- row -> {grid, rowgroup, table, treegrid}
+- rowgroup -> {grid, table, treegrid}
+- rowheader -> row
```

Comment:
./table-roles.html and/or ./grid-roles.html

Reply:
Done, please see aed8053a90073e291c93e45226bbe6fb2b5fb191